pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -718,131 +716,78 @@ func (service *SecurityPolicyService) deleteT1SecurityPolicy(spUid types.UID) er
-
+	// Further check if it's Default Project from vpcInfo
 	if !isDefaultProject {
-		if nsxSecurityPolicy != nil {
-			err = service.deleteNSXSecurityPolicy(nsxSecurityPolicy, &vpcInfo)
-			if err != nil {
-				log.Error(err, "Failed to delete NSX SecurityPolicy and rules in VPC", "nsxSecurityPolicyUID", spUID)
-				return err
-			}
-			err = service.applySecurityPolicyStore(nsxSecurityPolicy, *nsxRules, true)
-			if err != nil {
-				return err
-			}
-			log.Info("Successfully deleted NSX SecurityPolicy and rules only", "nsxSecurityPolicyUID", spUID)
-		}
-
-		err = service.deleteNSXSecurityPolicyGroupShare(*nsxGroups, *nsxShares, *nsxShareGroups, &vpcInfo)
-		// Ignore error here to make groups/shares to deleted in GC.
-		// Because NSX SecurityPolicy is deleted, it's unable to get SecurityPolicy from SecurityPolicyStore by requeuing the error.
+		isDefaultProject, err = service.vpcService.IsDefaultNSXProject(vpcInfo.OrgID, vpcInfo.ProjectID)
 		if err != nil {
-			log.Info("NSX SecurityPolicy is deleted, the associated groups and shares of the SecurityPolicy will be GC", "nsxSecurityPolicyUID", spUID)
-			return nil
+			log.Error(err, "Failed to check if NSX Project is default", "nsxProjectID", vpcInfo.ProjectID)
+			return err
 		}
-	} else {
-		// TODO: to refactor manipulateSecurityPolicyForDefaultProject to separate NSX security policy deletion and groups/shares deletion
-		_, err = service.manipulateSecurityPolicyForDefaultProject(nsxSecurityPolicy, *nsxGroups, *nsxShares, *nsxShareGroups, true, &vpcInfo)
+	}
+
+	if nsxSecurityPolicy != nil {
+		err = service.deleteNSXSecurityPolicy(nsxSecurityPolicy, &vpcInfo)
 		if err != nil {
 			log.Error(err, "Failed to delete NSX SecurityPolicy and rules in VPC", "nsxSecurityPolicyUID", spUID)
 			return err
 		}
-		err = service.applySecurityPolicyStore(nsxSecurityPolicy, *nsxRules, true)
+		err = service.applySecurityPolicyStore(nsxSecurityPolicy, nsxSecurityPolicy.Rules, true)
 		if err != nil {
 			return err
 		}
+		log.Info("Successfully deleted NSX SecurityPolicy and rules only", "nsxSecurityPolicyUID", spUID)
 	}
 
-	err = service.applyVPCGroupShareStore(*nsxGroups, *nsxShares, *nsxShareGroups, isDefaultProject)
+	if !isDefaultProject {
+		err = service.deleteNSXSecurityPolicyGroupShare(nsxGroups, nsxProjectShares, nsxProjectShareGroups, &vpcInfo)
+	} else {
+		err = service.deleteNSXSecurityPolicyGroupShareForDefaultProject(nsxGroups, nsxInfraShares, nsxInfraShareGroups, &vpcInfo)
+	}
+	// Ignore error here to make groups/shares to deleted in GC.
+	// Because NSX SecurityPolicy is deleted, it's unable to get SecurityPolicy from SecurityPolicyStore by requeuing the error.
+	// Also Ignore error that happens during GC process, because GC will be run in the next round, there is no need to return error to requeue.
+	if err != nil {
+		log.Info("NSX SecurityPolicy is deleted, the associated groups and shares of the SecurityPolicy will be GC", "nsxSecurityPolicyUID", spUID)
```

Comment:
It doesn't return the error but I think we'd better still print the error details.